via/static/scripts/video_player/components/YouTubeVideoPlayer.tsx

```diff
@@ -0,0 +1,174 @@
+import { useEffect, useRef, useState } from 'preact/hooks';
+
+import { loadYouTubeIFrameAPI } from '../utils/youtube';
+
+/**
+ * Playback state values.
+ *
+ * This is a copy of the `YT.PlayerState` global that is available when the
+ * YouTube IFrame API script has loaded, but it works in the test environment
+ * where we don't actually load that script.
+ *
+ * See https://developers.google.com/youtube/iframe_api_reference#Events.
+ */
+export const PlayerState = {
+  UNSTARTED: -1,
+  ENDED: 0,
+  PLAYING: 1,
+  PAUSED: 2,
+  BUFFERING: 3,
+  CUED: 4,
+};
+
+export type YouTubeVideoPlayerProps = {
+  /** ID of the YouTube video to load. */
+  videoId: string;
+
+  /**
+   * Whether the video is playing or paused.
+   *
+   * Note that even if this is set to `true` when the component is mounted, the
+   * browser may prevent automatic playback of the video until the user has
+   * interacted with the page.
+   */
+  play?: boolean;
+
+  /**
+   * Current play time of the video, expressed as a number of seconds since
+   * the start.
+   */
+  time?: number;
+
+  /**
+   * Callback invoked at a regular interval (eg. 1 second) when the timestamp
+   * of the video changes.
+   */
+  onTimeChanged?: (timestamp: number) => void;
+
+  /**
+   * Callback invoked when the playing/paused state of the video changes.
+   */
+  onPlayingChanged?: (playing: boolean) => void;
+};
+
+function isPlaying(state: YT.PlayerState) {
+  return state === PlayerState.PLAYING || state === PlayerState.BUFFERING;
+}
+
+/**
+ * Component that wraps the YouTube video player.
+ *
+ * See https://developers.google.com/youtube/iframe_api_reference.
+ */
+export default function YouTubeVideoPlayer({
+  onPlayingChanged,
+  onTimeChanged,
+  play = true,
+  videoId,
+  time,
+}: YouTubeVideoPlayerProps) {
+  const [loadError, setLoadError] = useState(null);
+  const encodedId = encodeURIComponent(videoId);
+  const encodedOrigin = encodeURIComponent(window.origin);
+  const playerURL = `https://www.youtube.com/embed/${encodedId}?enablejsapi=1&origin=${encodedOrigin}`;
```

Comment:
Could you add a comment with a link to the docs that explain how to build this URL?